packages/manager/apps/key-management-service/src/components/credential/credentialStatus/credentialStatus.component.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  ODS_CHIP_SIZE,
+  ODS_TEXT_COLOR_INTENT,
+  OdsChipAttribute,
+} from '@ovhcloud/ods-components';
+import { OsdsChip } from '@ovhcloud/ods-components/react';
+import { OkmsCredentialStatus } from '@/types/okmsCredential.type';
+
+export type KeyStatusProps = Omit<OdsChipAttribute, 'color'> & {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  state: OkmsCredentialStatus | (string & {});
+};
+
+export const CredentialStatus = ({ state, ...props }: KeyStatusProps) => {
+  const { t } = useTranslation('key-management-service/credential');
+
+  const { size, ...otherProps } = props;
+
+  let label = '';
+  let color: OdsChipAttribute['color'];
+
+  switch (state) {
+    case OkmsCredentialStatus.creating:
+      label = t('key_management_service_credential_status_creating');
+      color = ODS_TEXT_COLOR_INTENT.primary;
+      break;
+
+    case OkmsCredentialStatus.deleting:
+      label = t('key_management_service_credential_status_deleting');
+      color = ODS_TEXT_COLOR_INTENT.warning;
+      break;
+
+    case OkmsCredentialStatus.error:
+      label = t('key_management_service_credential_status_error');
+      color = ODS_TEXT_COLOR_INTENT.error;
+      break;
+
+    case OkmsCredentialStatus.expired:
+      label = t('key_management_service_credential_status_expired');
+      color = ODS_TEXT_COLOR_INTENT.default;
+      break;
+
+    case OkmsCredentialStatus.ready:
+      label = t('key_management_service_credential_status_ready');
+      color = ODS_TEXT_COLOR_INTENT.success;
+      break;
+
+    default:
+      label = state;
+      color = ODS_TEXT_COLOR_INTENT.default;
+      break;
+  }
```

Comment:
You can make an object 
```
const chipPropsByState: {[key in OkmsCredentialStatus]: {label: string; color: ODS_THEME_COLOR_INTENT}} = {
  [OkmsCredentialStatus.deleting]: {label: t('key...'), color: ODS_THEME_COLOR_INTENT.warning}
  // etc.
};

const props = chipPropsByState[state];

return (
  <OsdsChip color={props.color || ODS_THEME_COLOR_INTENT.default} {...otherProps}>{props.label || state}</OsdsChip> 
);
```

Reply:
And another idea to reduce the size of the component :
you could only put the key for the labels to be able to extract the constant outside the component then in the component you just have to apply the translation in the render

```
<OsdsChip>{t(labelKey) || state</OsdsChip>
```